### PR TITLE
fix: version deprecation failure false positive

### PR DESCRIPTION
**Pull request by sanpj2292**
## What are the changes introduced in this PR?

We are setting whole error object in destinationResponse which is triggering version deprecation alert(false-positive)

## What is the related Linear task?

Resolves INT-1617

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3104/test-report.html

File: test/integrations/destinations/gainsight_px/network.ts [hunk 1]
```diff
@@ -1,222 +1,514 @@
...
-                    "street": "",
-                    "postalCode": "",
-                    "continent": "",
-                    "regionName": "",
-                    "timeZone": "",
-                    "coordinates": {
-                        "latitude": 0.0,
-                        "longitude": 0.0
-                    }
+        httpsAgent: {
```
Comment (koladilip) [inline test/integrations/destinations/gainsight_px/network.ts:278]: do we need to keep the whole thing?

Comment (sanpj2292) [reply test/integrations/destinations/gainsight_px/network.ts:278]: It was only included to suggest that we got an AxiosError that was very long.

Comment (koladilip) [reply test/integrations/destinations/gainsight_px/network.ts:278]: Got it but do we need to test any code functionality?

Comment (sanpj2292) [reply test/integrations/destinations/gainsight_px/network.ts:278]: Yes, in a way. So we saw in our analysis that we had a keyword `upgrade required`, which is why the score of version deprecation increased leading to false-positive alerts.
In the bytes below, I think we would have some information about the keyword I was talking about hence put the information explicitly here

Through the test-case we are making sure that we don't put such information into payload when there is a failure from destination(that leads to `AxiosError`)

File: test/integrations/destinations/gainsight_px/network.ts [hunk 2]
```diff
@@ -1,222 +1,514 @@
...
-            method: 'GET',
+  },
+  // Axios Error
+  {
+    httpReq: {
+      url: 'https://api.aptrinsic.com/v1/users/myUId',
+      headers: { 'X-APTRINSIC-API-KEY': 'sample-api-key', 'Content-Type': 'application/json' },
+      method: 'GET',
+    },
+    httpRes: {
```
[the remark this answers is not in the document]
Comment (sanpj2292) [reply test/integrations/destinations/gainsight_px/network.ts:229]: As requested, I have trimmed down the response to a more readable one. Can you check and confirm, if the changes are looking good ?

**Review by koladilip — APPROVED** [review]

**Pull request merged**